### PR TITLE
Tests and fixes one more corner case of ToT x ToT evaluation.

### PR DESCRIPTION
**Pull request by bimalgaudel**

### CODE REVIEW

File: src/TiledArray/conversions/concat.h [hunk 1]
```diff
@@ -92,9 +92,8 @@ DistArray<Tile, Policy> concat(
   DistArray<Tile, Policy> result(*target_world, tr);
   const auto annot = detail::dummy_annotation(r);
   for (auto i = 0ul; i != arrays.size(); ++i) {
-    result.make_tsrexpr(annot).block(tile_begin_end[i].first,
-                                     tile_begin_end[i].second) =
-        arrays[i].make_tsrexpr(annot);
+    result(annot).block(tile_begin_end[i].first, tile_begin_end[i].second) =
```
Comment (evaleev) [inline src/TiledArray/conversions/concat.h:95]: The seems to undo the changes introduced in https://github.com/ValeevGroup/tiledarray/pull/461 ?

**Review by evaleev — APPROVED** [review]
